doc/inband-telemetry/SAI-Proposal-Advanced-Telemetry-draft.md

```diff
@@ -0,0 +1,740 @@
+     *
+     * @type sai_acl_field_data_t sai_uint16_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_INNER_ETHER_TYPE,
+
+    /**
+     * @brief Inner IP Protocol
+     *
+     * @type sai_acl_field_data_t sai_uint8_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_INNER_IP_PROTOCOL,
+
+    /**
+     * @brief Inner L4 Src Port
+     *
+     * @type sai_acl_field_data_t sai_uint16_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_INNER_L4_SRC_PORT,
+
+    /**
+     * @brief Inner L4 Dst Port
+     *
+     * @type sai_acl_field_data_t sai_uint16_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_ACL_ENTRY_ATTR_FIELD_INNER_L4_DST_PORT,
+
+    /**
+     * @brief Enable mirror on drop
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_MOD_ENABLE,
+
+    /**
+     * @brief Enable in-band network telemetry
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_INT_ENABLE,
+
+    /**
+     * @brief INT session ID
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_INT_SESSION,
+
+    /**
+     * @brief Enable packet postcard
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_POSTCARD_ENABLE,
+
+    /**
+     * @brief Telemetry flow sample percent within matched flow space
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 100
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_FLOW_SAMPLE_PERCENT,
+
+    /**
+     * @brief Report every packet for the matched flow
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default False
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_TELEMETRY_REPORT_ALL,
+
+} sai_acl_entry_attr_t;
+~~~
+
+## New Header `saitelemetry.h`
+
+### Data Structures and Enumerations
+
+~~~cpp
+/**
+ * @brief Queue alert report trigger attributes
+ */
+typedef enum _sai_telemetry_queue_alert_attr_t 
+{
+    /**
+     * @brief queue object ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_TELEMETRY_QUEUE_ALERT_ATTR_QUEUE_ID,
+
+    /**
+     * @brief queue depth threshold
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_TELEMETRY_QUEUE_ALERT_ATTR_QUEUE_DEPTH_THRESHOLD,
+
+    /**
+     * @brief queue latency threshold
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_TELEMETRY_QUEUE_ALERT_ATTR_QUEUE_LATENCY_THRESHOLD
+
+} sai_telemetry_queue_alert_attr_t;
+
+/**
+ * @brief INT config session attributes
+ */
+typedef enum _sai_telemetry_int_session_attr_t 
+{
+    /**
+     * @brief INT max hop count
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_TELEMETRY_INT_SESSION_ATTR_MAX_HOP_COUNT,
```

Comment:
what is the max_hop_count meta data in the session attribute?

Reply:
The maximum number of hops that are allowed to add their metadata to the packet. Added comments.